client/core/trade_simnet_test.go

```diff
@@ -87,15 +87,15 @@ func readWalletCfgsAndDexCert() error {
 	fp := filepath.Join
 	for _, client := range clients {
 		dcrw, btcw := client.dcrw(), client.btcw()
-		dcrw.config, err = config.Parse(fp(user.HomeDir, "dextest", "dcr", dcrw.name, "w-"+dcrw.name+".conf"))
+		dcrw.config, err = config.Parse(fp(user.HomeDir, "dextest", "dcr", dcrw.daemon, "w-"+dcrw.daemon+".conf"))
 		if err == nil {
-			btcw.config, err = config.Parse(fp(user.HomeDir, "dextest", "btc", "harness-ctl", btcw.name+".conf"))
+			btcw.config, err = config.Parse(fp(user.HomeDir, "dextest", "btc", "harness-ctl", btcw.daemon+".conf"))
 		}
 		if err != nil {
 			return err
 		}
-		dcrw.config["account"] = "default"
-		btcw.config["account"] = btcw.name
+		dcrw.config["account"] = dcrw.account
+		btcw.config["walletname"] = btcw.walletName
```

Comment:
I need to run this more often, sorry.  BTW I'm hoping at least one of the following possible errors you listed is resolved now:

https://github.com/decred/dcrdex/pull/589/files#diff-67aaad62146ed6da1f0f4291575bdd13R15-R18

Coin locking has improved a ton ofc, dcrwallet listunspent is fixed, and we're pulling only utxos from the funding account.  There's probably still bugs with all that, but it should be more robust than even a week ago.

Really confused about the `signrawtransaction` errors that show up randomly now though...

Reply:
Yeah, coin locking/unlocking has improved significantly. Just noticed an instance where dcrdex did not unlock a coin from a failed trade though and subsequent trading attempts with the coin return a `rpc error 36: coin <> is locked`.

A close examination of the logs below suggests that a couple failed trades were not revoked. The first match in the log is for `9ae7937ae39747044667978628da99231bebd3b9f5185ac1f52377cf5650e9c6` which ended at NewlyMatched (timestamp `00:51:12`).

Another match followed (`6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625`) which completed successfully (timestamp `00:54:18-00:54:46`).

Then 2 more matches (`3227781362fe63e9efde569a2fd590c76814f614d3284920ae94c5ea6c6da411` and `a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9`) ended at NewlyMatched and MakerSwap (timestamps `00:54:48-00:54:54` and `00:54:57-00:55:05`) respectively.

Then between `00:55:29 and 01:05:29`, the first match (`9ae7937a`) got revoked, although the revoke_match request could not be sent to the clients because they'd long disconnected. But that's all in my log regarding a revocation, which would explain the `coin is locked` server errors I started getting at some point.

<details>
<summary>dcrdex logs</summary>

```
2020-08-08 00:51:06.038 [TRC] MKT: Received order 8fa87deae70f261f31578f7b8763fb5a04cc13997255fc8fbd61d41185403e5f at 2020-08-08 00:51:06.038 +0000 UTC
2020-08-08 00:51:06.122 [TRC] MKT: Received order da482da19194eb15bd9e2865baa09e2dfbbbdfde2ad4b2fa32c185a6d4f2fc38 at 2020-08-08 00:51:06.122 +0000 UTC
2020-08-08 00:51:12.000 [TRC] MKT: Preimage request sent for order da482da19194eb15bd9e2865baa09e2dfbbbdfde2ad4b2fa32c185a6d4f2fc38
2020-08-08 00:51:12.000 [TRC] MKT: Preimage request sent for order 8fa87deae70f261f31578f7b8763fb5a04cc13997255fc8fbd61d41185403e5f
2020-08-08 00:51:12.000 [TRC] MKT: Good preimage received for order da482da19194eb15bd9e2865baa09e2dfbbbdfde2ad4b2fa32c185a6d4f2fc38: 4a443d973a0dd5939fc0cc7818e330a0be44e50580a1f1d921f3282e675d1f9a
2020-08-08 00:51:12.001 [TRC] MKT: Good preimage received for order 8fa87deae70f261f31578f7b8763fb5a04cc13997255fc8fbd61d41185403e5f: 31659b5c4177c1b82c4a22ae800274cafb4f5f81ef3de57f3557a5115693b120
2020-08-08 00:51:12.003 [INF] MKT: Collected 2 valid order preimages, missed 0. Commit checksum: ae2b645c01dcd59ea8faa66006bb20d053deecfd926eef82e21bf86cc8183caf
2020-08-08 00:51:12.003 [INF] MKT: Matching complete for market dcr_btc epoch 266141311: 1 matches (0 partial fills), 1 completed OK (not booked), 1 booked, 1 unbooked, 0 failed
2020-08-08 00:51:12.007 [DBG] MKT: Negotiating 1 matches for epoch 266141311:6000
2020-08-08 00:51:12.008 [WRN] SWAP: Unable to determining optimal fee rate for "dcr", using fallback of 10. Err: -32603: no bucket with the minimum required success percentage found
2020-08-08 00:51:12.010 [DBG] SWAP: Optimal fee rate for "btc": 6
2020-08-08 00:51:12.011 [DBG] SWAP: Negotiate: sending 'match' ack request to user 954b5da2a022fde4945080f88b61c1a906a23fbcfe548be8b795932aeaa24c24 for 1 matches
2020-08-08 00:51:12.011 [DBG] SWAP: Negotiate: sending 'match' ack request to user a4e38f1a4c4b3bd8222e90bfe1a26cf9a30a8e74219479c637f26329d4096a9b for 1 matches
2020-08-08 00:51:12.012 [DBG] SWAP: processMatchAcks: 'match' ack received from 954b5da2a022fde4945080f88b61c1a906a23fbcfe548be8b795932aeaa24c24 for 1 matches
2020-08-08 00:51:12.014 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from 954b5da2a022fde4945080f88b61c1a906a23fbcfe548be8b795932aeaa24c24 (maker=true) for match 9ae7937ae39747044667978628da99231bebd3b9f5185ac1f52377cf5650e9c6 (status NewlyMatched)
2020-08-08 00:51:12.014 [DBG] SWAP: processMatchAcks: 'match' ack received from a4e38f1a4c4b3bd8222e90bfe1a26cf9a30a8e74219479c637f26329d4096a9b for 1 matches
2020-08-08 00:51:12.015 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from a4e38f1a4c4b3bd8222e90bfe1a26cf9a30a8e74219479c637f26329d4096a9b (maker=false) for match 9ae7937ae39747044667978628da99231bebd3b9f5185ac1f52377cf5650e9c6 (status NewlyMatched)


2020-08-08 00:54:18.232 [TRC] MKT: Received order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d at 2020-08-08 00:54:18.232 +0000 UTC
2020-08-08 00:54:18.331 [TRC] MKT: Received order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99 at 2020-08-08 00:54:18.331 +0000 UTC
2020-08-08 00:54:24.000 [TRC] MKT: Preimage request sent for order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d
2020-08-08 00:54:24.000 [TRC] MKT: Preimage request sent for order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99
2020-08-08 00:54:24.001 [TRC] MKT: Good preimage received for order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99: 652dd91cc57c82ba925b6d6d587a95ea1e4ef20581465c503bf9d5c5c75baa1a
2020-08-08 00:54:24.001 [TRC] MKT: Good preimage received for order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d: 825d289786bbf28440ae2747a3a4d79ad996fad0a19fe92b1b28092a6eac605b
2020-08-08 00:54:24.002 [INF] MKT: Collected 2 valid order preimages, missed 0. Commit checksum: 8e5b3f59f83f5a975ae816351041575ba4f80298cb2c01bce7fd20a0dc95d059
2020-08-08 00:54:24.003 [INF] MKT: Matching complete for market dcr_btc epoch 266141343: 1 matches (0 partial fills), 1 completed OK (not booked), 1 booked, 1 unbooked, 0 failed
2020-08-08 00:54:24.007 [DBG] MKT: Negotiating 1 matches for epoch 266141343:6000
2020-08-08 00:54:24.007 [WRN] SWAP: Unable to determining optimal fee rate for "dcr", using fallback of 10. Err: -32603: no bucket with the minimum required success percentage found
2020-08-08 00:54:24.010 [DBG] SWAP: Optimal fee rate for "btc": 6
2020-08-08 00:54:24.011 [DBG] SWAP: Negotiate: sending 'match' ack request to user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:54:24.011 [DBG] SWAP: Negotiate: sending 'match' ack request to user 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for 1 matches
2020-08-08 00:54:24.012 [DBG] SWAP: processMatchAcks: 'match' ack received from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:54:24.014 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d (maker=true) for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625 (status NewlyMatched)
2020-08-08 00:54:24.014 [DBG] SWAP: processMatchAcks: 'match' ack received from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for 1 matches
2020-08-08 00:54:24.015 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 (maker=false) for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625 (status NewlyMatched)
2020-08-08 00:54:24.290 [DBG] SWAP: handleInit: 'init' received from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99
2020-08-08 00:54:24.291 [DBG] SWAP: step(user=fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d, match=6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625): ready for user 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 action, status NewlyMatched, next MakerSwapCast
2020-08-08 00:54:24.294 [DBG] SWAP: processInit: valid contract received at 2020-08-08 00:54:24.294 +0000 UTC from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, swapStatus NewlyMatched => MakerSwapCast
2020-08-08 00:54:24.295 [DBG] SWAP: processInit: sending contract 'audit' ack request to counterparty 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625
2020-08-08 00:54:29.620 [DBG] ASSET[btc]: Run: Processing new block 45ace13a6455ecea56671bb68ace942c227e3a55b034104a045b9c9ca17e0762
2020-08-08 00:54:29.620 [DBG] ASSET[btc]: Notifying 1 btc asset consumers of new block at height 261
2020-08-08 00:54:30.120 [DBG] SWAP: handleInit: 'init' received from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d
2020-08-08 00:54:30.120 [DBG] SWAP: step(user=4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662, match=6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625): ready for user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d action, status MakerSwapCast, next TakerSwapCast
2020-08-08 00:54:30.122 [DBG] SWAP: processInit: valid contract received at 2020-08-08 00:54:30.121 +0000 UTC from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, swapStatus MakerSwapCast => TakerSwapCast
2020-08-08 00:54:30.122 [DBG] SWAP: processInit: sending contract 'audit' ack request to counterparty fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625
2020-08-08 00:54:35.669 [DBG] ASSET[dcr]: Run: Processing new block 042c0c74febe9316c4408edd79faf5db3094199dd3420aa08470423749b26f8b
2020-08-08 00:54:35.669 [DBG] ASSET[dcr]: Notifying 1 dcr asset consumers of new block at height 264
2020-08-08 00:54:36.126 [DBG] SWAP: handleRedeem: 'redeem' received from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99
2020-08-08 00:54:36.126 [DBG] SWAP: step(user=fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d, match=6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625): ready for user 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 action, status TakerSwapCast, next MakerRedeemed
2020-08-08 00:54:36.133 [DBG] SWAP: processRedeem: valid redemption received at 2020-08-08 00:54:36.133 +0000 UTC from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, swapStatus TakerSwapCast => MakerRedeemed
2020-08-08 00:54:36.135 [DBG] SWAP: processRedeem: sending 'redeem' ack request to counterparty 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625
2020-08-08 00:54:36.137 [DBG] AUTH: Recorded order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d that has finished processing: user=4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662, time=1596848076137, target=<nil>
2020-08-08 00:54:36.162 [DBG] SWAP: handleRedeem: 'redeem' received from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, order c71b1526beabb8222e4106700f5e44fa3981ecd3f09b0aa406291c7e3a44f59d
2020-08-08 00:54:36.162 [DBG] SWAP: step(user=4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662, match=6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625): ready for user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d action, status MakerRedeemed, next MatchComplete
2020-08-08 00:54:41.189 [DBG] SWAP: processRedeem: valid redemption received at 2020-08-08 00:54:41.189 +0000 UTC from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625, swapStatus MakerRedeemed => MatchComplete
2020-08-08 00:54:41.191 [DBG] SWAP: processRedeem: sending 'redeem' ack request to counterparty fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625
2020-08-08 00:54:41.192 [DBG] AUTH: Recorded order a41110e76884fe103e27d5d2a050f5728cb968cfddd752c1a51b6954915cbc99 that has finished processing: user=fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d, time=1596848081192, target=<nil>
2020-08-08 00:54:41.669 [DBG] ASSET[dcr]: Run: Processing new block 2115b2c13bb2424315e2962482c2c68c416bb443977705f2aca00453bbe36240
2020-08-08 00:54:41.670 [DBG] ASSET[dcr]: Notifying 1 dcr asset consumers of new block at height 265
2020-08-08 00:54:46.620 [DBG] ASSET[btc]: Run: Processing new block 5113c862b734a7b34e8afccf38885917de5167ad2699d678b79b787b940c412f
2020-08-08 00:54:46.620 [DBG] ASSET[btc]: Notifying 1 btc asset consumers of new block at height 262
2020-08-08 00:54:46.621 [DBG] SWAP: Deleting completed match 6622fe589b2b46f28ffe15f2f344bbd625197840bb664828bfa8c85dcd47a625


2020-08-08 00:54:48.384 [TRC] MKT: Received order f15c0e183ae4573fe2c3dac7b2bbeb55464bed9f8b8806f8d7640948e455f3c0 at 2020-08-08 00:54:48.384 +0000 UTC
2020-08-08 00:54:48.447 [TRC] MKT: Received order 4fc559f5c193d4084496e59ae7a569d9a7a28daaaa0f477b19fcd4c4c5f5fd02 at 2020-08-08 00:54:48.447 +0000 UTC
2020-08-08 00:54:54.000 [TRC] MKT: Preimage request sent for order f15c0e183ae4573fe2c3dac7b2bbeb55464bed9f8b8806f8d7640948e455f3c0
2020-08-08 00:54:54.000 [TRC] MKT: Preimage request sent for order 4fc559f5c193d4084496e59ae7a569d9a7a28daaaa0f477b19fcd4c4c5f5fd02
2020-08-08 00:54:54.000 [TRC] MKT: Good preimage received for order f15c0e183ae4573fe2c3dac7b2bbeb55464bed9f8b8806f8d7640948e455f3c0: 7849400c1ca16ef732ab2f80fa26232a6ddbce4e9aef4c6a9f7c575d82319ced
2020-08-08 00:54:54.001 [TRC] MKT: Good preimage received for order 4fc559f5c193d4084496e59ae7a569d9a7a28daaaa0f477b19fcd4c4c5f5fd02: e65ec9d9ec3540e0d7c9085d26ed7cfdac67afed5565535310a9926e5e903479
2020-08-08 00:54:54.003 [INF] MKT: Collected 2 valid order preimages, missed 0. Commit checksum: eadea51c76bb67663faaed9fbaec0f120ed2622989a2401a690ead74d37fc398
2020-08-08 00:54:54.004 [INF] MKT: Matching complete for market dcr_btc epoch 266141348: 1 matches (0 partial fills), 1 completed OK (not booked), 1 booked, 1 unbooked, 0 failed
2020-08-08 00:54:54.007 [DBG] MKT: Negotiating 1 matches for epoch 266141348:6000
2020-08-08 00:54:54.008 [WRN] SWAP: Unable to determining optimal fee rate for "dcr", using fallback of 10. Err: -32603: no bucket with the minimum required success percentage found
2020-08-08 00:54:54.010 [DBG] SWAP: Optimal fee rate for "btc": 10
2020-08-08 00:54:54.012 [DBG] SWAP: Negotiate: sending 'match' ack request to user 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for 1 matches
2020-08-08 00:54:54.012 [DBG] SWAP: Negotiate: sending 'match' ack request to user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:54:54.013 [DBG] SWAP: processMatchAcks: 'match' ack received from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:54:54.014 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d (maker=false) for match 3227781362fe63e9efde569a2fd590c76814f614d3284920ae94c5ea6c6da411 (status NewlyMatched)
2020-08-08 00:54:54.014 [DBG] SWAP: processMatchAcks: 'match' ack received from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 for 1 matches
2020-08-08 00:54:54.015 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from 4b4e747bc2a5ad7d0baf6d2b42f6c98033edfdeb20020432f4c7bc7421571662 (maker=true) for match 3227781362fe63e9efde569a2fd590c76814f614d3284920ae94c5ea6c6da411 (status NewlyMatched)



2020-08-08 00:54:57.591 [TRC] MKT: Received order fe9e0c3f19ce8a4f93ef71a97cf4f609afb3f87129563f0849f116a0603a117a at 2020-08-08 00:54:57.591 +0000 UTC
2020-08-08 00:54:57.722 [TRC] MKT: Received order 46aa3325dac457c46c2342c55970e6e8938abf578b98939932eaced50171b019 at 2020-08-08 00:54:57.722 +0000 UTC
2020-08-08 00:55:00.000 [TRC] MKT: Preimage request sent for order 46aa3325dac457c46c2342c55970e6e8938abf578b98939932eaced50171b019
2020-08-08 00:55:00.000 [TRC] MKT: Preimage request sent for order fe9e0c3f19ce8a4f93ef71a97cf4f609afb3f87129563f0849f116a0603a117a
2020-08-08 00:55:00.001 [TRC] MKT: Good preimage received for order fe9e0c3f19ce8a4f93ef71a97cf4f609afb3f87129563f0849f116a0603a117a: 124bead42b6bd509a56d900e51068097e15faf533ec06ab7a72aa0aaeb4f8abc
2020-08-08 00:55:00.001 [TRC] MKT: Good preimage received for order 46aa3325dac457c46c2342c55970e6e8938abf578b98939932eaced50171b019: 0d9dace6e378fe1363c17229858efa252ddb4973a0c11ca672c25beff13b83e9
2020-08-08 00:55:00.002 [INF] MKT: Collected 2 valid order preimages, missed 0. Commit checksum: aa91db2166b6174a3d77e897fcd94c0519d9f3b9b226a37c1dcfb46e4c087cea
2020-08-08 00:55:00.003 [INF] MKT: Matching complete for market dcr_btc epoch 266141349: 1 matches (0 partial fills), 1 completed OK (not booked), 1 booked, 1 unbooked, 0 failed
2020-08-08 00:55:00.007 [DBG] MKT: Negotiating 1 matches for epoch 266141349:6000
2020-08-08 00:55:00.008 [WRN] SWAP: Unable to determining optimal fee rate for "dcr", using fallback of 10. Err: -32603: no bucket with the minimum required success percentage found
2020-08-08 00:55:00.011 [DBG] SWAP: Optimal fee rate for "btc": 10
2020-08-08 00:55:00.012 [DBG] SWAP: Negotiate: sending 'match' ack request to user 9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f for 1 matches
2020-08-08 00:55:00.012 [DBG] SWAP: Negotiate: sending 'match' ack request to user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:55:00.013 [DBG] SWAP: processMatchAcks: 'match' ack received from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for 1 matches
2020-08-08 00:55:00.014 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d (maker=false) for match a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9 (status NewlyMatched)
2020-08-08 00:55:00.014 [DBG] SWAP: processMatchAcks: 'match' ack received from 9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f for 1 matches
2020-08-08 00:55:00.015 [DBG] SWAP: processMatchAcks: storing valid 'match' ack signature from 9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f (maker=true) for match a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9 (status NewlyMatched)
2020-08-08 00:55:00.289 [DBG] SWAP: handleInit: 'init' received from 9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f for match a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9, order fe9e0c3f19ce8a4f93ef71a97cf4f609afb3f87129563f0849f116a0603a117a
2020-08-08 00:55:00.289 [DBG] SWAP: step(user=9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f, match=a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9): ready for user fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d action, status NewlyMatched, next MakerSwapCast
2020-08-08 00:55:00.294 [DBG] SWAP: processInit: valid contract received at 2020-08-08 00:55:00.293 +0000 UTC from 9622665526c9d888a8db16fe76da16796958def53deae2d31a759b32a7ceb69f for match a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9, swapStatus NewlyMatched => MakerSwapCast
2020-08-08 00:55:00.294 [DBG] SWAP: processInit: sending contract 'audit' ack request to counterparty fa21acf013cecff681e6adaa7322dfd7598369221671c7781e4d21bd23d3f39d for match a0483209f9a3da14a6885b7ef8dcf9a4f932c4046ac3f6e9028d84ac20c368c9


2020-08-08 00:55:29.620 [DBG] SWAP: checkInaction(failMatch): swap 9ae7937ae39747044667978628da99231bebd3b9f5185ac1f52377cf5650e9c6 failing (maker fault = true) at NewlyMatched
2020-08-08 00:55:29.622 [DBG] AUTH: Recorded order da482da19194eb15bd9e2865baa09e2dfbbbdfde2ad4b2fa32c185a6d4f2fc38 that has finished processing: user=a4e38f1a4c4b3bd8222e90bfe1a26cf9a30a8e74219479c637f26329d4096a9b, time=1596848129622, target=<nil>
2020-08-08 00:55:29.625 [DBG] AUTH: user 954b5da2a022fde4945080f88b61c1a906a23fbcfe548be8b795932aeaa24c24 penalized for rule 1
2020-08-08 00:55:29.625 [INF] SWAP: revoke: sending the 'revoke_match' request to each client for match 9ae7937ae39747044667978628da99231bebd3b9f5185ac1f52377cf5650e9c6
2020-08-08 01:05:29.626 [INF] SWAP: revoke_match request failed for user a4e38f1a4c4b3bd8222e90bfe1a26cf9a30a8e74219479c637f26329d4096a9b (taker)
2020-08-08 01:05:29.626 [INF] SWAP: revoke_match request failed for user 954b5da2a022fde4945080f88b61c1a906a23fbcfe548be8b795932aeaa24c24 (taker)
```

</details>

Reply:
The `rpc error 36: coin <> is locked` errors were frustrating me too, but I realized that's because I had restarted the test and/or the btc/dcr wallet harnesses without nuking the dcrdex harness.  So the server still had those coins locked from a previous run, but the test didn't realize they were locked on the server.  I think this was expected behavior if the server/dcrdex harness isn't also nuked between runs.

Reply:
For additional context, another set of 4 trades (3 failures, 1 success) - all failed trades were revoked but I tried re-running the test before the last trade was revoked and got the coin is locked error. Waited a bit, confirmed from dcrdex logs that the last trade was revoked and the tests re-ran without the coin is locked error.

I guess only question is why I had a previous instance where some trades were not revoked. Could be the dcrdex binary I was running is outdated, so will keep an eye on this.

The `not enough to cover requested funds` error could still occur at any point but it'd more likely mean that the configured account has no funds, rather than the wallet not having funds.

Reply:
Noticed this incorrect log bit (always `(taker)`) btw:https://github.com/decred/dcrdex/blob/8b55845bf6bf4fb83c9cfd6ba537ea2fd7462d81/server/swap/swap.go#L2068-L2079

Reply:
I mostly only nuke and restart dcrdex harness without visiting the btc/dcr harnesses. Could it be the dcr/btc wallets still consider the coins locked from before dcrdex re-run? iirc, dcrdex on launch unlocks all coins to start afresh.

In any case, I got the coin is locked error again, so issue's still on master. This time, a failed trade also failed to revoke, match ended at MakerSwapCast.

Reply:
Unless the dcrdex psql db is nuked, it will relock order coins on startup. There's no way the server would have coins locked from non-existent orders.  I never get this in the wild or in interactive simnet tests anyway.

When you say failed to revoke, does that mean the server didn't revoke it after broadcast timeout, or did one or both of maker or taker fail to revoke it?  That's another thing I never see in practice, so repro is needed to be sure it's not just a test harness thing.  That said, it could very well be the recent "retiring inactive trade" function getting in the way.

Reply:
- iirc, the dcrdex harness script nukes the simnet test db before restarting the binary.
- re failed to revoke, the server did not revoke, at least there was no log message to that effect on dcrdex (for some failed trades anyways), something to watch more closely, I guess.